### PR TITLE
Workflow: Linear Solvers Specific CodeQL

### PR DESCRIPTION
**Pull request by csiefer2**
Until @achauphan can get the self-hosted stuff working  (see #12440), this serves as a stop-gap for the Tpetra-based Linear Solvers portions of Trilinos for CodeQL analysis.  It runs using the GitHub default runner and takes 1.5 hours when I tested it on the muelu/Trilinos fork. It only builds Tpetra-based packages up through MueLu.

### CODE REVIEW

**Review by cgcgcg — APPROVED** [review]
This looks fine to me. Not sure what this will actually flag.

**Review by cgcgcg — APPROVED** [review]

**Comment by achauphan** [comment]
@csiefer2 this is unfortunate timing as I just saw this after merge, but we did just stand-up a few experimental self-hosted runners last week that you should be able to see [here](https://github.com/trilinos/Trilinos/settings/actions/runners) where we've been testing the new AutoTester2 system.

I'd be curious to see the performance differences for this on a bigger machine. At the moment, if you'd like to test this out in an open PR on one of those runners to get some metrics, I don't see any issue with that. You can use the `gcc-8.3.0_serial` label for the workflow file to run on those self-hosted runners.

For long term, it would be up to @sebrowne to decide if we have enough compute resources left over to dedicate to this workflow once we begin moving more and more of our Jenkins PR tests to GitHub actions as will be losing some of our machines in the next few months.

**Pull request merged**
